pkg/kademlia/kademlia.go

```diff
@@ -185,12 +183,22 @@ func (k *Kademlia) SetBootstrapNodes(nodes []pb.Node) { k.bootstrapNodes = nodes
 // Bootstrap contacts one of a set of pre defined trusted nodes on the network and
 // begins populating the local Kademlia node
 func (k *Kademlia) Bootstrap(ctx context.Context) error {
+	defer k.bootstrapFinished.Release()
+
+	k.lookups.Add(1)
+	defer k.lookups.Done()
+
 	if len(k.bootstrapNodes) == 0 {
-		return BootstrapErr.New("no bootstrap nodes provided")
```

Comment:
do we not care about not having a bootstrap node ?

Reply:
I'm not sure what to do about this. I mean we do for non-bootstrap peers, but don't care for bootstrap peer. I'm not sure whether there's a better way to handle this. I mean it could also check for the node type.